test/test-TabTracker.js

```diff
@@ -232,6 +232,35 @@ exports.test_TabTracker_refresh = function*(assert) {
   checkLoadUnloadReasons(assert, pingData, loadReasons, unloadReasons);
 };
 
+exports.test_TabTracker_action_pings = function*(assert) {
+  let pingData = {};
+  let userEventPromise = new Promise(resolve => {
+    function observe(subject, topic, data) {
+      if (topic === "user-action-event") {
+        pingData = JSON.parse(data);
+        Services.obs.removeObserver(observe, "user-action-event");
+        resolve();
+      }
+    }
+    Services.obs.addObserver(observe, "user-action-event");
+  });
+
+  let eventData = {
+    msg: {
+      data: {
+        source: "topsites",
+        action_position: 3,
+        event: "click",
+        page: "newtab"
+      }
+    }
+  };
+  app._handleUserEvent("NEW_USER_EVENT", eventData);
+
+  yield userEventPromise;
+  assert.deepEqual(eventData.msg.data, pingData, "We receive the expected ping data.");
```

Comment:
we should test for the presence of other properties in the pingData, such as: action, page, tab id, locale, client id.